### PR TITLE
modulesync: enable future parser tests

### PR DESCRIPTION
**Pull request by domcleal**

### CODE REVIEW

**Comment by domcleal** [comment]
Test failures fixed once #30 is merged.

**Comment by ekohl** [comment]
:+1: on green tests.

File: manifests/params.pp [hunk 1]
```diff
@@ -10,7 +10,7 @@
       } else {
         $root = '/srv/tftp'
       }
-      if $::operatingsystemrelease =~ /^8/ {
+      if versioncmp($::operatingsystemrelease, '8') >= 0 {
```
Comment (ekohl) [inline manifests/params.pp]: I now realize this may break Ubuntu 12.04 because old was = 8.*, now it's >= 8.

Comment (domcleal) [inline manifests/params.pp]: Well-spotted!  I'll perhaps check the operating system is Debian too?

Comment (ekohl) [inline manifests/params.pp]: I think that'd be good.

**Comment by domcleal** [comment]
I've changed the fixing commit, as I looked further down params.pp and noticed we also do a string comparison in the 'RedHat' case statement.  I realised the failure was just because the operatingsystemrelease fact was uninitialised in the test and is naturally a string, rather than there being a problem with the comparison.

**Comment by ekohl** [comment]
Maybe a good candidate to start using our facts gem.

**Comment by domcleal** [comment]
@ekohl yep, good point.  I've converted it to r-p-f, with the exception of the Amazon Linux example as facts aren't available for it, and it's not listed in our supported OSes.

I noticed that we can't reconfigure the TFTP root for Debian-based OSes, we ought perhaps to deploy or edit /etc/default/tftpd-hpa.

File: spec/classes/init_spec.rb [hunk 2]
```diff
@@ -1,108 +1,98 @@
...
+                       when 'Debian'
+                         'tftpd-hpa'
+                       end
+
+        should contain_package(tftp_package).with({
+          :ensure => 'installed',
+          :alias  => 'tftp-server',
+        })
+
+        if facts[:operatingsystem] == 'Debian' && facts[:operatingsystemrelease] == '8.0'
```
Comment (ekohl) [inline spec/classes/init_spec.rb]: Maybe check if the release starts with 8. in case the facts get updated to 8.1?

Comment (domcleal) [inline spec/classes/init_spec.rb]: Fixed, thanks.

**Comment by mmoll** [comment]
thanks!

**Pull request merged**